crates/cdk/src/mint/builder.rs

```diff
@@ -249,49 +249,105 @@ impl MintBuilder {
         };
 
         let settings = payment_processor.get_settings().await?;
-
-        let settings: Bolt11Settings = settings.try_into()?;
-
-        if settings.mpp {
-            let mpp_settings = MppMethodSettings {
-                method: method.clone(),
-                unit: unit.clone(),
-            };
-
-            let mut mpp = self.mint_info.nuts.nut15.clone();
-
-            mpp.methods.push(mpp_settings);
-
-            self.mint_info.nuts.nut15 = mpp;
+        let method_str = method.to_string();
+
+        // Handle bolt11 methods
+        if method_str == "bolt11" {
```

Comment:
I would prefer to use match on the enum then else if on the string if we can. It ensures we match all patterns including ones we can add in the future.